src/intrinsics.cpp

```diff
@@ -305,9 +305,13 @@ static Value *emit_unbox(Type *to, const jl_cgval_t &x, jl_value_t *jt)
     if (jt == (jl_value_t*)jl_bool_type)
         return builder.CreateZExt(builder.CreateTrunc(builder.CreateLoad(p), T_int1), to);
 
-    if (!x.isboxed) // stack has default alignment
+    if (x.isboxed)
+        return builder.CreateAlignedLoad(p, 16); // julia's gc gives 16-byte aligned addresses
```

Comment:
On 64bit I believe we garentee 16-byte alignment for GC allocated memory for all objects except singleton (although you don't really need to load from it...)

On 32bits, we garentee at least 4-byte alignment for objects smaller than 8 bytes, for anything larger than or equal to 8 bytes, we should have 16-byte alignment. Object with a non-zero size no larger than 4 bytes should also always be 8-byte aligned.....